lib/sshutils/scp/scp.go

```diff
@@ -349,6 +354,22 @@ func (cmd *command) sendFile(r *reader, ch io.ReadWriter, fileInfo FileInfo) err
 // serveSink executes file uploading, when a remote server sends file(s)
 // via scp
 func (cmd *command) serveSink(ch io.ReadWriter) error {
+	// Validate that if directory mode flag was sent, the target is an actual
+	// directory.
+	if cmd.Flags.DirectoryMode {
+		if len(cmd.Flags.Target) != 1 {
+			return trace.BadParameter("invalid number of targets")
```

Comment:
improve error message, what is the valid number? what targets were given, how user can fix the error